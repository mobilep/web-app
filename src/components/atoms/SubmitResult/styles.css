.SubmitResult {
	background: var(--white);
	color: var(--grey3);
	width: 100%;
	display: flex;
	flex-direction: column;
	justify-content: space-around;
	text-align: center;
	box-sizing: border-box;
}

.SubmitResult-content {
	padding-top: 10px;
	height: 100%;
	max-width: 570px;
}

.SubmitResult-title {
	font-size: 30px;
	font-weight: var(--semiBold);
}

.SubmitResult-message {
	margin-top: 20px;
	white-space: pre-line;
}

.SubmitResult-secondary-button + button {
	margin-left: 10px;
}

@media (max-width: 765px) {
	.SubmitResult-content {
		width: 100%;
		padding-bottom: 15px;
	}
}
