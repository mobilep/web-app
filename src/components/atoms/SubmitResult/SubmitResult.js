import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import './styles.css';

const SubmitResult = ({
	buttonLabel,
	className = '',
	icon,
	message,
	onButtonClick = () => {},
	title,
}) => {
	return (
		<div className={`SubmitResult ${className}`}>
			<div className={`SubmitResult-content`}>
				<div>{icon}</div>
				<div>
					<h3 className='SubmitResult-title'>
						{title}
					</h3>
					<div className='SubmitResult-message'>
						{message}
					</div>
				</div>
				<div>
					<Button onClick={onButtonClick}>
						{buttonLabel}
					</Button>
				</div>
			</div>
		</div>
	);
};

SubmitResult.propTypes = {
	buttonLabel: PropTypes.string.isRequired,
	className: PropTypes.string,
	icon: PropTypes.element,
	message: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
	onButtonClick: PropTypes.func.isRequired,
	onSecondaryButtonClick: PropTypes.func,
	secondaryButton: PropTypes.bool,
	secondaryButtonLabel: PropTypes.string,
	title: PropTypes.string,
};

export default SubmitResult;
