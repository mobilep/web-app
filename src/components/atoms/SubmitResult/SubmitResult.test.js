import React from 'react';
import { shallow } from 'enzyme';

import SubmitResult from './SubmitResult';

describe('Atom - SubmitResult', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<SubmitResult
				buttonLabel='Button label'
				icon={<div />}
				message='Message'
				onButtonClick={() => {}}
				title='Title'
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
