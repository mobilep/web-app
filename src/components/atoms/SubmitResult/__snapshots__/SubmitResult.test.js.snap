// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - SubmitResult Should render 1`] = `
<div
  className="SubmitResult "
>
  <div
    className="SubmitResult-content"
  >
    <div>
      <div />
    </div>
    <div>
      <h3
        className="SubmitResult-title"
      >
        Title
      </h3>
      <div
        className="SubmitResult-message"
      >
        Message
      </div>
    </div>
    <div>
      <Button
        onClick={[Function]}
      >
        Button label
      </Button>
    </div>
  </div>
</div>
`;
